### ISSUE TITLE
Do you get similar or the same result as the original paper?

### ISSUE THREAD

**Issue by lihk11** [rassilon712/Neural_Color_Transfer_Incomplete#1]
Hi, there~
Thanks for your implementation!
I tried to run one of the notebooks(NCT_modified), but the results didn't go very well. 
How about yours? Or maybe i do something wrong?

**Comment by rassilon712** [comment]
Hello 
We implemented the Neural Color Transfer and modified some of parameters but we also didn't get all fine results like paper.

We were trying to fix some of codes but, our team is guessing that the patch match is somewhat working not perfectly.

Also, the processing time is so much slower than the original paper because we implemented on Pytorch not CUDA exactly.

For now some results are similar with the paper and we are not making progress further from that implementation.

If you make better results please share with us and if we made further progress, I will update on my git.

Thanks.

**Comment by lihk11** [comment]
Thanks for your reply! 
Could you point out which pictures you run that reach similar result with the paper?  I think 
we can start with them.

**Comment by rassilon712** [comment]
Sorry for the late reply. We had hard time finding some good images we got.
Our Input Image is 
![Input](https://user-images.githubusercontent.com/15609545/54282220-e0ea0480-45de-11e9-810f-8d8467c0b60e.jpg)
and the style image is
![Style](https://user-images.githubusercontent.com/15609545/54282236-e8111280-45de-11e9-862c-a8c90724de85.jpg)

And the result from those images are here. The images are from layer 5 to 1 below.
![img5S](https://user-images.githubusercontent.com/15609545/54282066-8355b800-45de-11e9-9765-4071bf432e19.png)
![img4S](https://user-images.githubusercontent.com/15609545/54282069-851f7b80-45de-11e9-8bf6-7dd66ab3c9ac.png)
![img3S](https://user-images.githubusercontent.com/15609545/54282074-8781d580-45de-11e9-8c9b-c1f701736da5.png)
![img2S](https://user-images.githubusercontent.com/15609545/54282080-894b9900-45de-11e9-9192-af2c2ed814d6.png)
![img1S](https://user-images.githubusercontent.com/15609545/54282082-8badf300-45de-11e9-8b8b-9deeab193846.png)

Also we are trying to change some Markdown on front with results. But it is not completely implemented so we have lot of TODO.

**Comment by lihk11** [comment]
Thanks for your reply! 
I have implemented my own version, and i think i got a similar result for the original paper.
My best results:
![image](https://user-images.githubusercontent.com/4299373/54345118-599f9e00-467d-11e9-96b1-b8d46c2a9ad2.png)
![image](https://user-images.githubusercontent.com/4299373/54345121-5c01f800-467d-11e9-8ae7-1432a2324e02.png)
![image](https://user-images.githubusercontent.com/4299373/54345130-61f7d900-467d-11e9-9cb1-eac67f83e1bc.png)

As for the picture you given, here are my results(5 to 1):
![image](https://user-images.githubusercontent.com/4299373/54343814-91591680-467a-11e9-8adb-c1ab8687c954.png)
![image](https://user-images.githubusercontent.com/4299373/54343816-94540700-467a-11e9-9821-922e9e87b45f.png)
![image](https://user-images.githubusercontent.com/4299373/54343829-98802480-467a-11e9-98b1-68e09fcbc1c8.png)
![image](https://user-images.githubusercontent.com/4299373/54343830-9ae27e80-467a-11e9-8869-c9ac816b98f2.png)
![image](https://user-images.githubusercontent.com/4299373/54343836-9e760580-467a-11e9-8d65-d0973257458c.png)

**Comment by lihk11** [comment]
I just checked your version. I think the parameter "radius" of FastGuidedFilter is not set properly. It could be larger (eg. 16). I am not very sure, cuz there are a lot of differences between our implements.

**Comment by nieguangyang** [comment]
Hi @rassilon712 , 
I'm currently working on the NCT model with tensorflow and keras based on your implementation.
Here is output of my implementation
![neural_color_transfer](https://user-images.githubusercontent.com/23467887/57994833-9eb6d500-7af1-11e9-9828-c1d5bbadefec.jpg)

It's pretty well on your given picture, but when it comes to some other pictures, the output are barely satisfied.

The biggest reason that I think might be the limit of capability of VGG19. It seems require very high semantic similarity to perform color transfer.

Now I am trying to replace VGG19 with other more semantic-oriented model.

**Comment by nieguangyang** [comment]
> Thanks for your reply!
> I have implemented my own version, and i think i got a similar result for the original paper.
> My best results:
> ![image](https://user-images.githubusercontent.com/4299373/54345118-599f9e00-467d-11e9-96b1-b8d46c2a9ad2.png)
> ![image](https://user-images.githubusercontent.com/4299373/54345121-5c01f800-467d-11e9-8ae7-1432a2324e02.png)
> ![image](https://user-images.githubusercontent.com/4299373/54345130-61f7d900-467d-11e9-9cb1-eac67f83e1bc.png)
> 
> As for the picture you given, here are my results(5 to 1):
> ![image](https://user-images.githubusercontent.com/4299373/54343814-91591680-467a-11e9-8adb-c1ab8687c954.png)
> ![image](https://user-images.githubusercontent.com/4299373/54343816-94540700-467a-11e9-9821-922e9e87b45f.png)
> ![image](https://user-images.githubusercontent.com/4299373/54343829-98802480-467a-11e9-98b1-68e09fcbc1c8.png)
> ![image](https://user-images.githubusercontent.com/4299373/54343830-9ae27e80-467a-11e9-8869-c9ac816b98f2.png)
> ![image](https://user-images.githubusercontent.com/4299373/54343836-9e760580-467a-11e9-8d65-d0973257458c.png)

Hi @lihk11 ,
I'm currently working on NCT too and my version got bad results for the pictures in the paper.
I think the biggest problem is loss d instead radius of fast guided filter. I've tried loss d exactly as showed in the paper and the modified version of rassilon712, non of them got decent output for pictures in the paper, can I have a look at your implementation of loss d?

**Comment by rassilon712** [comment]
Hello, sorry for late reply. Me and my co-worker are busy on doing other projects so, this NCT progress is delayed for moments.
For the implementation of our code is based on version 1 of the paper. We need to change some loss functions as the version 2 of the paper changed some loss terms.
Did you check the version 2 of the paper and followed their terms? Also, my work needs some tuning for GuidedFilter, even though we have had some tests.

**Comment by nieguangyang** [comment]
> Hello, sorry for late reply. Me and my co-worker are busy on doing other projects so, this NCT progress is delayed for moments.
> For the implementation of our code is based on version 1 of the paper. We need to change some loss functions as the version 2 of the paper changed some loss terms.
> Did you check the version 2 of the paper and followed their terms? Also, my work needs some tuning for GuidedFilter, even though we have had some tests.

Thanks for reply~
I think most changes in v2 paper is about the loss d. It contains a factor 4 ** (L - 1). Although it still not work well for me. Weirdly, when I change it to 4 ** (5 - L), got better result. Therefore, now it is confusing for me...
Looking forward to your later improvement and hope the paper's authors can release the source code soon, there are so many tricky parts!

**Comment by rassilon712** [comment]
Yes, I also think that there are some tricks in code that they didn't mention in paper. We will work further when we have enough time and if you get some remarkable results, I hope to share the results with us !

**Comment by nieguangyang** [comment]
Sure, I will! :)

**Comment by Ssakura-go** [comment]
> urther when we have enough time and if you get some remarkable results, I hope to share the results with us !

Hi,  could you please share the code you run with me?
Recently I focus on color transfer task and also learn a lots from this paper. Now I want to run out the results by myself, however, I got the wrong code and I can't find out the available implement of it at present.
